### PR TITLE
Result.py: Modify docstring definition of diffs

### PR DESCRIPTION
**Pull request by aptrishu**
It modifies docstring param definition of diffs
as it was confusing. There should be only one
diff object attached with a particular filename.
Also, it modifies Result.apply code to make it
more readable.

Fixes https://github.com/coala/coala/issues/3034

### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [x] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [x] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

File: coalib/results/Result.py [hunk 1]
```diff
@@ -208,8 +208,8 @@ def apply(self, file_dict: dict):
         :param file_dict: A dictionary containing all files with filename as
                           key and all lines a value. Will be modified.
         """
-        for filename in self.diffs:
-            file_dict[filename] = self.diffs[filename].modified
+        for filename, diff in self.diffs.items():
+            file_dict[filename] = diff.modified
```
Comment (Makman2) [inline coalib/results/Result.py]: own commit :D

Comment (aptrishu) [inline coalib/results/Result.py]: Yeah :P

Comment (aptrishu) [inline coalib/results/Result.py]: damn should have made 2 commits.. >_<

File: coalib/results/Result.py [hunk 2]
```diff
@@ -60,8 +60,8 @@ def __init__(self,
             A message which may help the user find out why this result was
```
Comment (Makman2) [inline coalib/results/Result.py]: Could you maybe leave out the "definition" of "docstring definition" in your commit message?
Actually we have a component in coala called `DocstyleDefinition` which is highly related to this term and defines a set of properties that identify a docstring in code (like the delimiters, in python these would be `"""`).

File: coalib/results/Result.py (hunk 1 above)
Comment (aptrishu) [inline coalib/results/Result.py]: made 2 commits.

**Comment by hemangsk** [comment]
ack c68530b167cf38aaef643836ce3e6b81a822502d

File: coalib/results/Result.py (hunk 2 above)
Comment (Makman2) [inline coalib/results/Result.py]: nvm

File: coalib/results/Result.py [hunk 3]
```diff
@@ -208,8 +208,8 @@ def apply(self, file_dict: dict):
         :param file_dict: A dictionary containing all files with filename as
```
Comment (Makman2) [inline coalib/results/Result.py]: this time commit message does really need change, your shortlog is to general^^

You did simplify code, so you could just say `Result: Simplify` or `Result.apply(): Simplify`.

\+ The issue reference `Fixes xxx` is wrong, please remove it^^ The other commit does fix the issue :+1:

\+
> [...] and it follows the param definition of diffs.

I don't understand that :O

Comment (aptrishu) [inline coalib/results/Result.py]: the docstring change commit fixes that issue right ?

Comment (Makman2) [inline coalib/results/Result.py]: yes :+1:

Comment (aptrishu) [inline coalib/results/Result.py]: Ok, made the changes. :)

File: coalib/results/Result.py [hunk 4]
```diff
@@ -60,8 +60,8 @@ def __init__(self,
             A message which may help the user find out why this result was
             yielded.
         :param diffs:
-            A dictionary with filenames as key and a sequence of ``Diff``
-            objects associated with them as values.
+            A dictionary with filename as key and `Diff` object
```
Comment (Makman2) [inline coalib/results/Result.py]: what about double backticks? I think we should use those, I've read somewhere that those do link to the actual classes or so in RST. Not really sure, but it's anyway more consistent then^^

File: coalib/results/Result.py [hunk 5]
```diff
@@ -137,8 +137,8 @@ def from_values(cls,
...
```
Comment (Makman2) [inline coalib/results/Result.py]: same here

**Comment by Makman2** [comment]
ack 55044bd

**Comment by Makman2** [comment]
unack c68530b

**Comment by Makman2** [comment]
one last shot and it's ready :)

**Comment by jayvdb** [comment]
ack f389b76 580a9a7

**Comment by jayvdb** [comment]
@rultor merge

**Comment by aptrishu** [comment]
rebased. it can be merged now. @jayvdb  :)

**Comment by Mixih** [comment]
reack ae3c0ca 77a3f88

**Comment by Mixih** [comment]
@rultor merge

**Pull request merged**